src/dpsvalidationservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/dpsvalidationservice/GetValidOpenDFCMCase.java

```diff
@@ -6,31 +6,86 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.regex.Pattern;
+
 
 @RestController
 public class GetValidOpenDFCMCase {
     private static final Logger logger = LogManager.getLogger(GetValidOpenDFCMCase.class);
 
+    /**
+     * getValidOpenDFCMCase
+     * <p>
+     * WARNING: Do not modify input parameters. This operation must replicate legacy
+     * system exactly.
+     * <p>
+     * Note: In the event of a unexpected parameter patter this methods will return an HTTP status
+     * code of 200. THIS IS EXPECTED AND WHAT THE LEGACY SYSTEM DOES. PLEASE DO NOT
+     * CHANGE THIS. An error is indicated to the calling system by the -2 int
+     * value in the response structure.
+     *
+     * @param driversLicense
+     * @param surcode
+     */
     @RequestMapping(value = "/getValidOpenDFCMCase",
             produces = {"application/xml"},
             method = RequestMethod.GET)
     @ApiOperation(value = "Generates application callback endpoints for DPS Validation Service", notes = "", tags = {"DPSValidationService"})
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Successful operation")})
     public String getValidOpenDFCMCase(@ApiParam(value = "driversLicense", required = true) @RequestParam(value = "driversLicense", required = true) String driversLicense, @ApiParam(value = "surcode", required = true) @RequestParam(value = "surcode", required = true) String surcode) {
-        /* TODO implement the logic as in the webmethods */
+        /* TODO implement the logic as in the webmethods with the ORDS pieces in  */
         logger.info("getValidOpenDFCMCase is called");
+
         String integer_ = "2"; // This will be changed later to accept the int value from the ORDS
         String caseDesc = "ROUTINE - PROFESSIONAL"; // This will be changed later to accept the caseDesc value from the ORDS
 
-        return String.format("<GetValidOpenDFCMCase>\n" +
-                "<int>%s</int>\n" +
-                "<caseDesc>%s</caseDesc>\n" +
-                "</GetValidOpenDFCMCase>\n", integer_, caseDesc);
+        boolean match_driversLicense = Pattern.matches("[0-9]{7}", driversLicense);
+        boolean match_surcode = Pattern.matches("^[a-zA-Z&-.]{0,3}", surcode);
+
+        if (!match_driversLicense) {
+            logger.fatal("The drivers license pattern is not matching the expectation.");
```

Comment:
Suggested small change to the fatal error msg "Invalid parameter format" and then in the log say something like "Invalid driverLicence format". This way we don't give away too much information about why the formatting failed to the caller who may be malicious but we report to the log what actually was going wrong.

Reply:
messgae corrected